src/tangents.jl

```diff
@@ -408,20 +411,24 @@ end
 Returns the unique zero element of the tangent space of `x`.
 It is an error for the zero element of the tangent space of `x` to be represented by
 anything other than that which this function returns.
+
+Internally, `zero_tangent` calls `zero_tangent_internal`, which handles different types of inputs differently.
+`zero_tangent_internal` has two variants:
+1. For `isbitstype` types, `zero_tangent_internal` takes one argument. 
+2. Otherwise, `zero_tangent_internal` takes another argument which is an `IdDict`, which solves the issue of circular references
+and also allows more efficient handling of aliased objects.
```

Comment:
```suggestion
2. Otherwise, `zero_tangent_internal` takes another argument which is an `IdDict`, which
handles both circular references and aliasing correctly.
```